### DISCUSSION TITLE
[Feature] Container argument to lower `--storage.tsdb.retention.time`

### DISCUSSION THREAD

**Discussion by jcpunk** [prometheus/prometheus#11604] [Q&A]
### Proposal

Use case:
For my environment, I've got constrained storage.  It would be helpful if there was a way I could tell the prometheus container to set `--storage.tsdb.retention.time` to a much lower time (my target is 27 hours).

Benefit to the project: Folks running a series of leaf containers can lower the the retention time to avoid storing metrics beyond their requirements.  This will result in lower storage usage and thus less cost when folks pay per consumption.

**Comment by roidelapluie** [comment]
This can be changed, I do not understand the issue.

**Comment by jcpunk** [comment]
I'm not showing a way to change the retention time via the config file.  In order to set `--storage.tsdb.retention.time` right now, a new downstream container needs to be built.  Unless I've missed something.

**Comment by roidelapluie** [comment]
The flags can be defined when you start the container. It does not need a rebuild.

**Reply by jcpunk** [reply]
I may be out of my depth here, but I'm not seeing obvious documentation on how to do that....
